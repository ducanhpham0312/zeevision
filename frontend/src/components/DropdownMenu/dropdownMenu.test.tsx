import "@testing-library/jest-dom";
import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuItem,
  DropdownMenuTrigger,
} from "./DropdownMenu";

function renderDropdownMenu() {
  return render(
    <div style={{ pointerEvents: "auto" }}>
      <DropdownMenu>
        <DropdownMenuTrigger>Options</DropdownMenuTrigger>
        <DropdownMenuContent>
          <DropdownMenuItem>Item 1</DropdownMenuItem>
        </DropdownMenuContent>
      </DropdownMenu>
    </div>,
  );
}

describe("DropdownMenu Component", () => {
  let triggerButton: HTMLElement;

  beforeEach(() => {
    renderDropdownMenu();
    triggerButton = screen.getByText("Options");
  });

  it("render snapshot correctly", async () => {
    const { baseElement } = renderDropdownMenu();
    // Initial snapshot before interaction
    expect(baseElement).toMatchSnapshot();

    // Wait for the menu items to be in the document
    await userEvent.click(triggerButton);
    await waitFor(() => {
      expect(screen.getByText("Item 1")).toBeInTheDocument();
    });
    // Add a debug statement here
    screen.debug();
    // Take a snapshot after the menu items are confirmed to be present
    expect(baseElement).toMatchSnapshot();
  });

  it("renders DropdownMenu and its items", async () => {
    // Check if the trigger is rendered
    expect(screen.getByText("Options")).toBeInTheDocument();

    // Check if the menu items are rendered - Initially not visible
    expect(screen.queryByText("Item 1")).not.toBeInTheDocument();

    // Wait for the menu items to be in the document
    await userEvent.click(triggerButton);
    await waitFor(() => expect(screen.getByText("Item 1")).toBeInTheDocument());
  });

  it("opens and closes the dropdown on trigger click", async () => {
    // Initially, the menu is closed
    expect(screen.queryByText("Item 1")).not.toBeInTheDocument();

    // Open the menu
    await userEvent.click(triggerButton);
    await waitFor(() => expect(screen.getByText("Item 1")).toBeInTheDocument());

    // Close the menu
    await userEvent.click(triggerButton);
    await waitFor(() =>
      expect(screen.queryByText("Item 1")).not.toBeInTheDocument(),
    );
  });
});
