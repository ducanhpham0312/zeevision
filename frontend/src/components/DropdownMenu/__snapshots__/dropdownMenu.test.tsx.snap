// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu Component render snapshot correctly 1`] = `
<body>
  <div>
    <div
      style="pointer-events: auto;"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        data-state="closed"
        id="radix-:r0:"
        type="button"
      >
        Options
      </button>
    </div>
  </div>
  <div>
    <div
      style="pointer-events: auto;"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        data-state="closed"
        id="radix-:r2:"
        type="button"
      >
        Options
      </button>
    </div>
  </div>
</body>
`;

exports[`DropdownMenu Component render snapshot correctly 2`] = `
<body
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      style="pointer-events: auto;"
    >
      <button
        aria-controls="radix-:r1:"
        aria-expanded="true"
        aria-haspopup="menu"
        data-state="open"
        id="radix-:r0:"
        type="button"
      >
        Options
      </button>
    </div>
  </div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      style="pointer-events: auto;"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        data-state="closed"
        id="radix-:r2:"
        type="button"
      >
        Options
      </button>
    </div>
  </div>
  <div
    data-radix-popper-content-wrapper=""
    dir="ltr"
    style="position: fixed; left: 0px; top: 0px; transform: translate(0px, 4px); min-width: max-content; --radix-popper-available-width: 0px; --radix-popper-available-height: -4px; --radix-popper-anchor-width: 0px; --radix-popper-anchor-height: 0px; --radix-popper-transform-origin: 50% 0px;"
  >
    <div
      aria-labelledby="radix-:r0:"
      aria-orientation="vertical"
      class="text-popover-foreground data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 z-50 min-w-[8rem] overflow-hidden rounded-md border bg-popover p-1 shadow-md data-[state=closed]:animate-out data-[state=open]:animate-in"
      data-align="center"
      data-orientation="vertical"
      data-radix-menu-content=""
      data-side="bottom"
      data-state="open"
      dir="ltr"
      id="radix-:r1:"
      role="menu"
      style="outline: none; --radix-dropdown-menu-content-transform-origin: var(--radix-popper-transform-origin); --radix-dropdown-menu-content-available-width: var(--radix-popper-available-width); --radix-dropdown-menu-content-available-height: var(--radix-popper-available-height); --radix-dropdown-menu-trigger-width: var(--radix-popper-anchor-width); --radix-dropdown-menu-trigger-height: var(--radix-popper-anchor-height); pointer-events: auto;"
      tabindex="-1"
    >
      <div
        class="focus:text-accent-foreground relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors focus:bg-hover data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
        data-orientation="vertical"
        data-radix-collection-item=""
        role="menuitem"
        tabindex="-1"
      >
        Item 1
      </div>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;
